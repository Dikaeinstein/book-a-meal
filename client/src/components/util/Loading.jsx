import React from 'react';
import PropTypes from 'prop-types';
import { Col, Row } from 'react-materialize';
import '../../static/sass/materialize.scss';

/**
 * displays a loading progress bar
 *
 * @param {Object} props object with property, text
 *
 * @returns {JSX} JSX element
 */
const Loading = ({ children, text }) => (
  <section style={{ margin: '0 auto', textAlign: 'center' }}>
    <Row>
      <Col s={12}>
        <strong style={{ display: 'block', marginBottom: '1rem' }}>
          {text}
        </strong>
        {children}
      </Col>
    </Row>
  </section>
);

Loading.propTypes = {
  text: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
};

export default Loading;
