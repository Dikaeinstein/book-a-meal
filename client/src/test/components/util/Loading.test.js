import React from 'react';
import { shallow } from 'enzyme';
import Loading from '../../../components/util/Loading';

describe('<Loading />', () => {
  it('should match snapshot', () => {
    const tree = shallow(<Loading text="test"><div>test loader</div></Loading>);
    expect(tree).toMatchSnapshot();
  });
});
