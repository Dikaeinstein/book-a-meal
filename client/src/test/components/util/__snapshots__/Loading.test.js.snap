// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should match snapshot 1`] = `
<section
  style={
    Object {
      "margin": "0 auto",
      "textAlign": "center",
    }
  }
>
  <Row>
    <Col
      s={12}
    >
      <strong
        style={
          Object {
            "display": "block",
            "marginBottom": "1rem",
          }
        }
      >
        test
      </strong>
      <div>
        test loader
      </div>
    </Col>
  </Row>
</section>
`;
